### PR TITLE
Create extension spec

### PR DESCRIPTION
**Pull request by alessandratrapani**
Comments on https://github.com/catalystneuro/ndx-facemap-motionsvd/issues/2

### CODE REVIEW

File: src/pynwb/ndx_facemap_motionsvd/__init__.py [hunk 1]
```diff
@@ -21,11 +21,12 @@
...
+# `@register_class("MotionSVDSeries", "ndx-facemap-motionsvd")`
+MotionSVDSeries = get_class("MotionSVDSeries", "ndx-facemap-motionsvd")
+MotionSVDMasks = get_class("MotionSVDMasks", "ndx-facemap-motionsvd")
 
 # NOTE: `widgets/tetrode_series_widget.py` adds a "widget"
-# attribute to the TetrodeSeries class. This attribute is used by NWBWidgets.
+# attribute to the MotionSVDSeries class. This attribute is used by NWBWidgets.
 # Delete the `widgets` subpackage or the `tetrode_series_widget.py` module
 # if you do not want to define a custom widget for your extension neurodata
 # type.
```
Comment (weiglszonja) [inline src/pynwb/ndx_facemap_motionsvd/__init__.py]:
```suggestion

```

File: src/pynwb/ndx_facemap_motionsvd/__init__.py [hunk 2]
```diff
@@ -21,11 +21,12 @@
 # TODO: Define your classes here to make them accessible at the package level.
 # Either have PyNWB generate a class from the spec using `get_class` as shown
 # below or write a custom class and register it using the class decorator
-# `@register_class("TetrodeSeries", "ndx-facemap-motionsvd")`
-TetrodeSeries = get_class("TetrodeSeries", "ndx-facemap-motionsvd")
+# `@register_class("MotionSVDSeries", "ndx-facemap-motionsvd")`
+MotionSVDSeries = get_class("MotionSVDSeries", "ndx-facemap-motionsvd")
+MotionSVDMasks = get_class("MotionSVDMasks", "ndx-facemap-motionsvd")
 
 # NOTE: `widgets/tetrode_series_widget.py` adds a "widget"
```
Comment (weiglszonja) [inline src/pynwb/ndx_facemap_motionsvd/__init__.py]:
```suggestion

```

File: src/pynwb/tests/test_motionsvdseries.py [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+        self.assertEqual(motionsvd_series.name, "MotionSVDSeries")
+        self.assertEqual(motionsvd_series.description, "description")
+        np.testing.assert_array_equal(motionsvd_series.data, data)
+        self.assertEqual(motionsvd_series.rate, 1000.0)
+        self.assertEqual(motionsvd_series.starting_time, 0)
+        self.assertEqual(motionsvd_series.motion_masks, motion_masks)
+
+
+class TestMotionSVDSeriesSimpleRoundtrip(TestCase):
```
Comment (weiglszonja) [inline src/pynwb/tests/test_motionsvdseries.py:68]: Can you also add a roundtrip test using `NWBH5IOFlexMixin` based on https://github.com/catalystneuro/ndx-turner-metadata/blob/c577aa29d00a2c8afecc9f5d535cb68839c2b129/src/pynwb/tests/test_labmetadata_extension.py#L62C46-L62C63 ?

**Comment by weiglszonja** [comment]
Looks great @alessandratrapani, needs a README.md but otherwise looks good.

**Comment by alessandratrapani** [comment]
Hey @weiglszonja, I made few changes because the extension was not working properly, and I realize it when implementing the test that you requested, so thank you very much for that! 
The create_extension_spec.py was creating two objects that could not retain the info about the namespace and the new datatype. I re-defined the two object in classes in facemap_motionsvd.py, as  @CodyCBakerPhD suggested.
[Here](https://catalystneuro-processing.s3.us-east-2.amazonaws.com/higley-lab-to-nwb/facemap_motionsvd.nwb) a nwb file generated as in the README example.

**Review by weiglszonja — APPROVED** [review]
Thanks @alessandratrapani, this looks good to me.

**Pull request merged**